llvm/test/Transforms/ThinLTOBitcodeWriter/split-typed-decl.ll

```diff
@@ -0,0 +1,12 @@
+;; Generating bitcode files with split LTO modules should not crash if there are
+;; typed declarations in sources.
+
+; RUN: opt -passes=elim-avail-extern --thinlto-bc --thinlto-split-lto-unit -o - %s

```

Comment:
Rather than run elim-avail-extern, you could just change the IR of the test to directly contain the output of that pass (a declaration with type metadata), and only run the bitcode writer.